### PR TITLE
Add more testing for configuration and running server

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Adds testing for generating the `elasticsearch.Client` configuration. Adds testing for `BindAddress`. Adds tests for `runServer`.

Adds `make check` to ensure that all files are correct. Run `make check` by Jenkins.

## Why is it important?

To catch errors before they are committed.

## Checklist

- [X] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [X] I have added tests that prove my fix is effective or that my feature works
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

File: cmd/fleet/server_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	bc := NewBulkCheckin()
+	ct := NewCheckinT(bc, ba, pm)
+	et := NewEnrollerT(cfg)
+
+	router := NewRouter(ctx, sv, ct, et)
+
+	go func() {
+		err = runServer(ctx, router, cfg)
+	}()
+	<-time.After(500 * time.Millisecond)
```
Comment (aleksmaus) [inline cmd/fleet/server_test.go:42]: non-deterministic

File: go.mod [hunk 2]
```diff
@@ -3,14 +3,13 @@ module fleet
...
 require (
-	github.com/a-h/generate v0.0.0-20190312091541-e59c34d33fb3 // indirect
-	github.com/aleksmaus/generate v0.0.0-20201201195936-7bace6a09854 // indirect
 	github.com/dgraph-io/ristretto v0.0.3
 	github.com/elastic/beats/v7 v7.10.0
 	github.com/elastic/go-elasticsearch/v8 v8.0.0-20200728144331-527225d8e836
 	github.com/elastic/go-ucfg v0.8.3
 	github.com/gofrs/uuid v3.3.0+incompatible
 	github.com/google/go-cmp v0.4.0
+	github.com/google/uuid v1.1.2-0.20190416172445-c2e93f3ae59f
```
Comment (aleksmaus) [inline go.mod:12]: we already have github.com/gofrs/uuid
, no need for the google one

**Review by aleksmaus — APPROVED** [review]
remove the google/uuid.
otherwise LGTM

File: cmd/fleet/server_test.go (hunk 1 above)
Comment (blakerouse) [reply cmd/fleet/server_test.go:47]: cleaned this up to ensure it waits for it to actual stop, then check the error

File: go.mod (hunk 2 above)
Comment (blakerouse) [reply go.mod]: done

File: cmd/fleet/server_test.go (hunk 1 above)
Comment (aleksmaus) [reply cmd/fleet/server_test.go:47]: thank you!

Comment (aleksmaus) [reply cmd/fleet/server_test.go:47]: still can cancel too early on a slow machine ;-)

File: internal/pkg/config/output_test.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+	"github.com/elastic/beats/v7/libbeat/common/transport/tlscommon"
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestToESConfig(t *testing.T) {
```
Comment (ruflin) [inline internal/pkg/config/output_test.go:21]: :wohoo:, ES tests.

**Pull request merged**